scripts/dev/generate_config_types.py

```diff
@@ -0,0 +1,152 @@
+"""This script auto-generates the `qutebrowser/config/configcontainer.py` file."""
+
+import pathlib
+import textwrap
+from typing import TYPE_CHECKING, NoReturn, Union
+from collections.abc import Mapping, Iterator
+
+from qutebrowser.config import configdata
+from qutebrowser.config.configdata import Option
+
+
+if TYPE_CHECKING:
+    from typing_extensions import assert_never
+else:
+
+    def assert_never(value) -> NoReturn:
+        raise AssertionError(f"Expected code to be unreachable, but got: {repr(value)}")
+
+
+NestedConfig = dict[str, Union[Option, "NestedConfig"]]
+
+
+def make_nested_config(config: Mapping[str, Option]) -> NestedConfig:
+    """The original configdata.yml defines nested keys using `.`s in a flat tree.
+
+    This function returns a new dict where options are grouped and nested, e.g.
+    `'auto_save.session': Option(...)` -> `{'auto_save': {'session': Option(...)}}`
+    """
+    result = {}
+    for key, value in config.items():
+        parts = key.split(".")
+        current = result
+        for part in parts[:-1]:
+            current = current.setdefault(part, {})
+        current[parts[-1]] = value
+    return result
+
+
+def generate_config_types(config_data: Mapping[str, Option]) -> Iterator[str]:
+    """Generate the `ConfigContainer` dataclass and all nested types."""
+    yield "# SPDX-FileCopyrightText: Florian Bruhin (The Compiler) <mail@qutebrowser.org>"
+    yield "#"
+    yield "# SPDX-License-Identifier: GPL-3.0-or-later"
+    yield ""
+    yield "# DO NOT EDIT THIS FILE DIRECTLY!"
+    yield "# It is autogenerated by running:"
+    yield "#   $ python3 scripts/dev/src2asciidoc.py"
+    yield "# vim: readonly:"
```

Comment:
ah this results in a lint warning saying to use `.editorconfig`, can we disable the lint for this file? or should I just remove this directive?